src/utils/stark.ts

```diff
@@ -49,3 +49,15 @@ export function formatSignature(sig?: Signature): string[] {
     return [];
   }
 }
+
+export function compileCalldata(args: RawArgs): Calldata {
```

Comment:
Some comments here could be nice, what's the difference between this one and the one in `Contract`?

Reply:
Would it make sense to have JSDocs type comments? For example:

```javascript
  /**
   * Signs a transaction with the starknet private key and returns the signature
   *
   * @param invocation the invocation object containing:
   * - contractAddress - the address of the contract
   * - entrypoint - the entrypoint of the contract
   * - calldata - (defaults to []) the calldata
   * - signature - (defaults to []) the signature
   * @param abi (optional) the abi of the contract for better displaying
   *
   * @returns signature
   */
   ```